### ISSUE TITLE
confirm that content was preserved and replicated correctly

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/infrastructure-integration-test#248]
We of course have tests that create SDR objects, and at least in the case of `create_preassembly_image_spec.rb`, a successful test somewhat implies that preservation is working correctly, because it should not be possible to complete accessioning of v1 and v2 without v1 completing `preservationIngestWF` successfully.

Still, we don't directly check to see that the content is sitting in Moab storage roots as expected (the part implied by successful completion of `preservationIngestWF` and `accessionWF`), and we don't check to see that content was replicated to cloud endpoints successfully (which is not implied by successful completion of any workflows, and thus not even implicitly checked by the integration tests, because it is triggered asynchronously completely within pres cat upon successful completion of the `update-catalog` step in `preservationIngestWF`).

As a sort of MVP approach, I'd suggest augmenting `create_preassembly_image_spec.rb` to check that the content was replicated correctly.  This seems to me like it'd provide the most value for the given work, since this is currently not checked at all by the infrastructure integration tests.

two possible approaches:
1. have the integration test SSH into a pres cat worker VM, and have it execute a small script via Rails runner that would 1) synchronously run replication audit for the new druid, and 2) query to confirm that all the expected parts exist in the pres cat database with an `ok` status.  There should be some helpful guidance in this wiki entry, though feel free to ask @jmartin-sul to pair if that'd be helpful at all:  https://github.com/sul-dlss/preservation_catalog/wiki/Investigating-a-druid-with-replication-errors
2. expose the aforementioned pres cat functionality via REST: a controller method that kicks off replication audit (async), and a controller method that exposes replication info for a druid, which can be polled by the integration tests.  _Pros_: cleaner separation between the tests and the application, less knowledge of the inner workings of pres cat by the integration tests, and a couple of REST calls that might be usable by other systems in the future (e.g. indexing could ask for replication info for a druid).  _Cons_: probably a bit more implementation work in total, and would require minting and sharing a private (pres cat) API token for infra integration tests.

Implementing this ticket would eliminate some of the manual testing follow up required by e.g. https://github.com/sul-dlss/preservation_robots/issues/310, https://github.com/sul-dlss/preservation_catalog/pull/1589, and https://github.com/sul-dlss/preservation_catalog/pull/1614.

**Comment by jmartin-sul** [comment]
Another possibility: pres cat records successful replication to a cloud endpoint with an entry in the DSA event log, and the event log on the Argo object page is checked for the eventual appearance of an event for each cloud endpoint for each version.

I think that would happen when `ResultsRecorderJob` determines that all parts for a druid/version/endpoint combo have been replicated.  Which I would suggest adding as a check between the current [lines 29 and 30 of that job's `#perform` method](https://github.com/sul-dlss/preservation_catalog/blob/4dcfdf4ace23af797da06b41ccdc2b9a141f5a32/app/jobs/results_recorder_job.rb#L29-L30).

I would also lean slightly towards adding a simple expectation about the events getting recorded to https://github.com/sul-dlss/preservation_catalog/blob/4dcfdf4ace23af797da06b41ccdc2b9a141f5a32/spec/jobs/integration_spec.rb#L52 (but don't feel strongly, maybe that's overkill, and keeping that integration test relatively uncluttered is nice)

At first glance I think I like this third approach the most.  It adds app behavior that's useful for real world debugging, it's probably a simpler change than either of the other options, there's no reaching under the hood on pres cat and running scripts, and it requires no new token provisioning.

I would suggest recording... the size of the replicated zip in each event?  Which should provide a nice useful sanity check, both for the infra integration test and in real world usage... not sure what other useful info might go in the event.  That seems fine to me as a start.

Will mull it over a bit more, but will likely close this ticket in favor of two smaller tickets: one in pres cat for the event logging, and one in this repo, blocked by the pres cat ticket, to look for the appearance of replication events on the Argo object show page.

**Comment by jmartin-sul** [comment]
closing in favor of https://github.com/sul-dlss/preservation_catalog/issues/1685 and https://github.com/sul-dlss/infrastructure-integration-test/issues/249
